src/Features/CSharp/Portable/CodeRefactorings/ConvertToRecord/CSharpConvertToRecordRefactoringProvider.cs

```diff
@@ -0,0 +1,791 @@
+                                {
+                                    // if it is too complex to remove (param validation, error handling, side effects, not everything assigned)
+                                    // we won't delete, and give a warning
+                                    // TODO: if parameter defaults in constructor, move those to primary constructor
+                                    // TODO: Change warning string
+                                    modifiedMembers.Add(
+                                        constructor.WithAdditionalAnnotations(WarningAnnotation.Create("Placeholder Constructor warning")));
+                                }
+                                continue;
+                            }
+
+                            if (constructorParamTypes.Length == 1 &&
+                                constructorParamTypes.First() == classDeclaration.Identifier.ToString())
+                            {
+                                // TODO: Check to see whether it's worth deleting
+                            }
+
+                            // TODO: Can potentially refactor statements which initialize properties with a simple expression
+                            // (not using local variables) to be moved into the this args
+                            var thisArgs = SyntaxFactory.ArgumentList(
+                                SyntaxFactory.SeparatedList(
+                                    Enumerable.Repeat(
+                                        SyntaxFactory.Argument(
+                                            SyntaxFactory.LiteralExpression(SyntaxKind.DefaultLiteralExpression)),
+                                        propertiesToMove.Length)));
+
+                            // TODO: look to see if there is already initializer (base or this)
+
+                            var modifiedConstructor = constructor
+                                .WithInitializer(SyntaxFactory.ConstructorInitializer(SyntaxKind.ThisConstructorInitializer, thisArgs));
+
+                            modifiedMembers.Add(modifiedConstructor);
+                            continue;
+                        }
+
+                    case OperatorDeclarationSyntax op:
+                        {
+                            // keep track of equality overloads so we can potentially remove them later
+                            var opKind = op.OperatorToken.Kind();
+                            if (opKind == SyntaxKind.EqualsEqualsToken)
+                            {
+                                equals = op;
+                            }
+                            else if (opKind == SyntaxKind.ExclamationEqualsToken)
+                            {
+                                notEquals = op;
+                            }
+
+                            break;
+                        }
+
+                    case MethodDeclarationSyntax method:
+                        {
```

Comment:
these individual case blocks get huge and difficult to undertand (especially with all the continuing and whatnot).  better to just extract out a helepr function that you pass this method to that is responsible just for this work).